src/main/java/io/antmedia/streamsource/StreamFetcher.java

```diff
@@ -168,140 +166,188 @@ public Result prepare(AVFormatContext inputFormatContext) {
 
 		@Override
 		public void run() {
 
-			if (isJobRunning.compareAndSet(false, true)) {
+			setThreadActive(true);
+			long lastPacketTime = 0, firstPacketTime = 0, bufferDuration = 0;
 
-				setThreadActive(true);
-				AVFormatContext inputFormatContext = null;
-				AVPacket pkt = null;
-				try {
-					inputFormatContext = new AVFormatContext(null); // avformat.avformat_alloc_context();
-					pkt = avcodec.av_packet_alloc();
-					//logger.info("before prepare");
-					Result result = prepare(inputFormatContext);
+			AVPacket pkt = null;
+			String packetWriterJobName = null;
+			try {
+				inputFormatContext = new AVFormatContext(null); // avformat.avformat_alloc_context();
+				pkt = avcodec.av_packet_alloc();
+				//logger.info("before prepare");
+				Result result = prepare(inputFormatContext);
 
-					if (result.isSuccess()) {
 
-						muxAdaptor = MuxAdaptor.initializeMuxAdaptor(null,true, scope);
+				if (result.isSuccess()) {
 
-						muxAdaptor.init(scope, stream.getStreamId(), false);
+					
+					muxAdaptor = MuxAdaptor.initializeMuxAdaptor(null,true, scope);
+					muxAdaptor.init(scope, stream.getStreamId(), false);
 
-						logger.info("{} stream count in stream {} is {}", stream.getStreamId(), stream.getStreamUrl(), inputFormatContext.nb_streams());
+					logger.info("{} stream count in stream {} is {}", stream.getStreamId(), stream.getStreamUrl(), inputFormatContext.nb_streams());
 
-						if(muxAdaptor.prepareInternal(inputFormatContext)) {
+					if(muxAdaptor.prepareInternal(inputFormatContext)) {
 
-							long currentTime = System.currentTimeMillis();
-							muxAdaptor.setStartTime(currentTime);
+						long currentTime = System.currentTimeMillis();
+						muxAdaptor.setStartTime(currentTime);
 
-							getInstance().startPublish(stream.getStreamId());
+						getInstance().startPublish(stream.getStreamId());
 
-							while (true) {
-								int ret = av_read_frame(inputFormatContext, pkt);
-								if (ret < 0) {
-									logger.info("cannot read frame from input context");
-									break;
-								}
-								streamPublished=true;
-								lastPacketReceivedTime = System.currentTimeMillis();
-
-								/**
-								 * Check that dts values are monotically increasing for each stream
-								 */
-								int packetIndex = pkt.stream_index();
-								if (lastDTS[packetIndex] >= pkt.dts()) {
-									pkt.dts(lastDTS[packetIndex] + 1);
-									//logger.warn("Correcting dts value to {}", pkt.dts());
-								}
-								lastDTS[packetIndex] = pkt.dts();
-								if (pkt.dts() > pkt.pts()) {
-									pkt.pts(pkt.dts());
-								}
-
-								if (lastPTS[packetIndex] >= pkt.pts()) {
-									pkt.pts(lastPTS[packetIndex] + 1);
-									//	logger.warn("Correcting pts value to {}", pkt.pts());
-								}
-								lastPTS[packetIndex] = pkt.pts();
+						if (bufferTime > 0) {
+							packetWriterJobName = scheduler.addScheduledJob(5, this);
```

Comment:
What is this 5 and why 5? Please define a final static int for it.

Reply:
Yeah OK. https://github.com/ant-media/Ant-Media-Server/commit/cef4dc9e1b87c6539100fbce7886a8897c756c94

We make 10ms because 1000 / 10 = 100 so that 100 packets can be handled with this period. Nowadays streams having 30fps have 80-90 packets in one second